numcodecs/zarr3.py

```diff
@@ -256,10 +256,10 @@ def compute_encoded_size(self, input_byte_length: int, chunk_spec: ArraySpec) ->
 class Shuffle(_NumcodecsBytesBytesCodec):
     codec_name = f"{CODEC_PREFIX}shuffle"
 
-    def __init__(self, **codec_config: dict[str, JSON]) -> None:
+    def __init__(self, **codec_config: JSON) -> None:
         super().__init__(**codec_config)
 
-    def evolve_from_array_spec(self, array_spec: ArraySpec) -> Self:
+    def evolve_from_array_spec(self, array_spec: ArraySpec) -> Shuffle:
```

Comment:
Becuase the method can explicitly return an instance of `Shuffle`, a sub-class of `Shuffle` would return `Shuffle` and not `Self`. Thinking about this, perhaps this should be `Shuffle | Self` since the method can also return `Self`.